miden-crypto/src/merkle/smt/full/leaf.rs

```diff
@@ -110,6 +110,15 @@ impl SmtLeaf {
             }
         }
 
+        let mut entries = entries;
+        entries.sort_by(|(key_1, _), (key_2, _)| cmp_keys(*key_1, *key_2));
+
+        for pair in entries.windows(2) {
+            if pair[0].0 == pair[1].0 {
+                return Err(SmtLeafError::DuplicateKeysInMultipleLeaf { key: pair[0].0 });
+            }
+        }
```

Comment:
These checks are currently done in `pairs_to_leaf()` function an din the `insert()` function. It may still make sense to move the checks here (to check these invariants during deserialization) - but then we should make sure we remove the no longer needed code these function.